CalibTracker/SiStripCommon/plugins/ShallowGainCalibration.cc

```diff
@@ -36,17 +34,17 @@ void ShallowGainCalibration::
 produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
   std::auto_ptr<std::vector<int> >            trackindex    ( new std::vector<int>            );
   std::auto_ptr<std::vector<unsigned int> >   rawid         ( new std::vector<unsigned int>   );
-  std::auto_ptr<std::vector<float>  >         localdirx     ( new std::vector<float>          );
-  std::auto_ptr<std::vector<float>  >         localdiry     ( new std::vector<float>          );
-  std::auto_ptr<std::vector<float>  >         localdirz     ( new std::vector<float>          );
+  std::auto_ptr<std::vector<double> >         localdirx     ( new std::vector<double>         );
+  std::auto_ptr<std::vector<double> >         localdiry     ( new std::vector<double>         );
+  std::auto_ptr<std::vector<double> >         localdirz     ( new std::vector<double>         );
```

Comment:
Hi @davidlange6 , yes there were. In order to obtain the **very** same results as in the previous version we had to move to floats. Our binning is very fine and we started seeing some rounding errors when making non linear combinations (ratios, multiplications, etc..) of the variables.

I am planning to review the code and see if we can revert to float by pre-processing some information, but this will take time and we would like to have these changes in 8.0